ngraph/core/src/op/gather.cpp

```diff
@@ -128,6 +128,172 @@ shared_ptr<Node> op::v1::Gather::clone_with_new_inputs(const OutputVector& new_a
     return make_shared<v1::Gather>(new_args.at(PARAMS), new_args.at(INDICES), new_args.at(AXIS));
 }
 
+NGRAPH_RTTI_DEFINITION(op::v7::Gather, "Gather", 7);
+static constexpr int64_t AXIS_NOT_SET_VALUE = std::numeric_limits<int64_t>::max();
```

Comment:
I don't like the approach with the variable `AXIS_NOT_SET_VALUE`. We should check that the input with `axes` is not connected in the places where the value of `axes` is needed.

Reply:
I copied this solution from the Gather-1. I think using of getter is preferable because `axis` can be < 0, if we will get it manually we will need to check and convert to positive in every place we use it. 

But I agree that this `AXIS_NOT_SET_VALUE` looks unsightly. 
What do you think about the following solution:  add `NODE_VALIDATION_CHECK` inside of `get_axis()` and remove variable `AXIS_NOT_SET_VALUE`?

Reply:
Just properly support case when axis is dynamic. After that the `AXIS_NOT_SET_VALUE` can be removed

Reply:
done